client/asset/btc/btc.go

```diff
@@ -875,36 +881,46 @@ func newRPCWallet(requester RawRequesterWithContext, cfg *BTCCloneCFG, walletCon
 	return &ExchangeWalletFullNode{btc}, nil
 }
 
-func newUnconnectedWallet(cfg *BTCCloneCFG, walletCfg *WalletConfig) (*baseWallet, error) {
+func validateBtcWalletConf(walletCfg *WalletConfig) error {
 	// If set in the user config, the fallback fee will be in conventional units
 	// per kB, e.g. BTC/kB. Translate that to sats/byte.
-	fallbackFeesPerByte := toSatoshi(walletCfg.FallbackFeeRate / 1000)
-	if fallbackFeesPerByte == 0 {
-		fallbackFeesPerByte = cfg.DefaultFallbackFee
+	walletCfg.FallbackFeeRatePerByte = toSatoshi(walletCfg.FallbackFeeRate / 1000)
+	if walletCfg.FallbackFeeRatePerByte == 0 {
+		return fmt.Errorf("fallback fee rate limit is smaller than smallest unit: %v",
+			walletCfg.FallbackFeeRate)
 	}
-	cfg.Logger.Tracef("Fallback fees set at %d %s/vbyte",
-		fallbackFeesPerByte, cfg.WalletInfo.UnitInfo.AtomicUnit)
-
 	// If set in the user config, the fee rate limit will be in units of BTC/KB.
 	// Convert to sats/byte & error if value is smaller than smallest unit.
-	feesLimitPerByte := cfg.DefaultFeeRateLimit
-	if feesLimitPerByte == 0 {
-		feesLimitPerByte = defaultFeeRateLimit
-	}
-	if walletCfg.FeeRateLimit > 0 {
-		feesLimitPerByte = toSatoshi(walletCfg.FeeRateLimit / 1000)
-		if feesLimitPerByte == 0 {
-			return nil, fmt.Errorf("fee rate limit is smaller than smallest unit: %v",
-				walletCfg.FeeRateLimit)
-		}
+	walletCfg.FeeRateLimitPerByte = toSatoshi(walletCfg.FeeRateLimit / 1000)
+	if walletCfg.FeeRateLimitPerByte == 0 {
+		return fmt.Errorf("fee rate limit is smaller than smallest unit: %v",
+			walletCfg.FeeRateLimit)
 	}
-	cfg.Logger.Tracef("Fees rate limit set at %d sats/byte", feesLimitPerByte)
 
 	redeemConfTarget := walletCfg.RedeemConfTarget
 	if redeemConfTarget == 0 {
-		redeemConfTarget = defaultRedeemConfTarget
+		return fmt.Errorf("redeem conf target can not be 0: %v",
+			walletCfg.RedeemConfTarget)
+	}
+
+	return nil
+}
+
+func newUnconnectedWallet(cfg *BTCCloneCFG, walletCfg *WalletConfig) (*baseWallet, error) {
+	err := validateBtcWalletConf(walletCfg)
+	if err != nil {
+		cfg.Logger.Tracef(err.Error())
+		// check which value failed, and add the default value to it
+		if walletCfg.FallbackFeeRatePerByte == 0 {
+			walletCfg.FallbackFeeRatePerByte = cfg.DefaultFallbackFee
+		}
+		if walletCfg.FeeRateLimitPerByte == 0 {
+			walletCfg.FeeRateLimitPerByte = cfg.DefaultFeeRateLimit
```

Comment:
If the previous value, `FallbackFeeRate`, failed validation, then `validateBtcWalletConf` would have returned before it set `walletCfg.FeeRateLimitPerByte`, which may have been valid, but it would go with the default instead.

I suggest were rework this a little, making `validateBtcWalletConf` not mutate the struct, and only applying the defaults to the fields that failed validation.

Reply:
I removed the early return from the validate method. 

I also noticed this values being re-applied on error were actually doing nothing, as they would be overwritten by the forms input, so I removed them.

Reply:
oh, actually they need to be overwritten on error, otherwise tests fail. 

Okay, undoing that

Reply:
> If we want default values if values not set can we set those first and then do validate and error if that errors?

What do you mean by set those value first? Right now the default values are populated from the `configOpts` and if changing its value on the form, it will not be the default values anymore.

> Isn't it correct to return error if values are wrong here? This happens when opening an existing wallet correct? We don't want to silently ignore what the user thinks they set, overwrite it and use some values they don't want?

In the UI an error will be thrown, when creating the wallet, but I agree with you about returning the error, the reason I am not returning it, is to keep the old behavior: 

Right now on the walletTypeSPV test it passes an empty wallet config struct, so it will fail if we return the errors here

```
	case walletTypeSPV:
		w, err := newUnconnectedWallet(cfg, &WalletConfig{})
		if err == nil {
			wallet = &ExchangeWalletFullNode{w}
			neutrinoClient := &tNeutrinoClient{data}
			wallet.node = &spvWallet{
				chainParams: &chaincfg.MainNetParams,
				wallet:      &tBtcWallet{data},
				cl:          neutrinoClient,
				tipChan:     make(chan *block, 1),
				chainClient: nil,
				acctNum:     0,
				txBlocks:    data.dbBlockForTx,
				checkpoints: data.checkpoints,
				log:         cfg.Logger.SubLogger("SPV"),
				loader:      nil,
			}
		}
```

Reply:
This looks correct now. But I don't think we need the trace log as it will log as error later.